src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Telemetry/ProjectTelemetryEvent.cs

```diff
@@ -45,5 +64,7 @@ public class ProjectTelemetryEvent : TelemetryEvent
         /// True, if project can be upgraded to PackageReference.
         /// </summary>
         public bool IsProjectPRUpgradable => (bool)base[IsPRUpgradable];
+
+        public string FullPath => (string)GetPiiData().Where(kv => kv.Key == nameof(FullPath)).First().Value;
     }
```

Comment:
Maybe not now but it looks bit perf issue by looking up key from IEnumerable loop. Maybe it needs little refactor to dictionary since [GetPiiData()](https://github.com/NuGet/NuGet.Client/blob/3b5c425bea75ba20cb6136e37944176c9e45c774/src/NuGet.Core/NuGet.Common/Telemetry/TelemetryEvent.cs#L75-L80) itself already has KeyValuePair values.  Also currently not many references of `GetPiiData()` so it looks possible change.

Reply:
This property looks like it's only used by a test. I think this property should just be removed. See also the comment I'm about to make on the test itself.